doctr/models/recognition/master/pytorch.py

```diff
@@ -0,0 +1,284 @@
+        input_shape: Tuple[int, int, int] = (3, 48, 160),
+        cfg: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__()
+
+        self.max_length = max_length
+        self.vocab = vocab
+        self.cfg = cfg
+        self.vocab_size = len(vocab)
+        self.num_heads = num_heads
+
+        self.feature_extractor = MAGCResnet(headers=headers)
+        self.seq_embedding = nn.Embedding(self.vocab_size + 1, d_model)  # One additional class for EOS
+
+        self.decoder = Decoder(
+            num_layers=num_layers,
+            d_model=d_model,
+            num_heads=num_heads,
+            dff=dff,
+            vocab_size=self.vocab_size,
+            maximum_position_encoding=max_length,
+        )
+        self.feature_pe = positional_encoding(input_shape[1] * input_shape[2], d_model)
+        self.linear = nn.Linear(d_model, self.vocab_size + 1)
+
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(m.weight, mode='fan_out', nonlinearity='relu')
+            elif isinstance(m, (nn.BatchNorm2d, nn.GroupNorm)):
+                nn.init.constant_(m.weight, 1)
+                nn.init.constant_(m.bias, 0)
+
+    def make_mask(self, target: torch.Tensor) -> torch.Tensor:
+        look_ahead_mask = create_look_ahead_mask(target.size(1))
+        target_padding_mask = create_padding_mask(target, self.vocab_size)  # Pad with EOS
+        combined_mask = target_padding_mask & look_ahead_mask
+        return torch.tile(combined_mask.permute(1, 0, 2), (self.num_heads, 1, 1))
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        target: Optional[List[str]] = None,
+        return_model_output: bool = False,
+        return_preds: bool = False,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        """Call function for training
+
+        Args:
+            x: images
+            target: list of str labels
+            return_model_output: if True, return logits
+            return_preds: if True, decode logits
+
+        Return:
+            A torch tensor, containing logits
+        """
+
+        # Encode
+        feature = self.feature_extractor(x, **kwargs)
+        b, c, h, w = (feature.size(i) for i in range(4))
+        feature = torch.reshape(feature, shape=(b, c, h * w))
+        feature = feature.permute(0, 2, 1)  # shape (b, h*w, c)
+        encoded = feature + self.feature_pe[:, :h * w, :]
+
+        if target is not None:
+            # Compute target: tensor of gts and sequence lengths
+            gt, seq_len = self.compute_target(target)
+            tgt_mask = self.make_mask(torch.from_numpy(gt))
+            # Compute logits
+            output = self.decoder(torch.from_numpy(gt), encoded, tgt_mask, None)
+            logits = self.linear(output)
+
+        else:
+            _, logits = self.decode(encoded)
+
+        out: Dict[str, torch.Tensor] = {}
+        out['out_map'] = logits
+
+        return out
```

Comment:
Could you check how the forward is implemented in the other archs ?

From what I read, the loss computation is not implemented, but still you have some lines using the "gt", what about the preds ?